tests/test_shipyard.py

```diff
@@ -0,0 +1,475 @@
+@file test_shipyard
+
+Holds the unit tests for shipyard.py, which is mainly PyQT interactions
+"""
+import sys
+import pytest
+from pytestqt import qtbot
+from imperium.shipyard import Window
+
+
+@pytest.fixture()
+def window(qtbot):
+    """ Before all for the tests, initializing a Window object """
+    window = Window()
+    qtbot.addWidget(window)
+    yield window
+
+
+def test_base_stats_init(window):
+    """ Tests for a base init of the Base Stats """
+    assert window is not None
+    assert window.spacecraft.tonnage == 100
+
+    # Base Stats
+    assert window.spacecraft.get_remaining_cargo() == 90
+    assert window.cargo_line_edit.text() == "90"
+
+    assert window.fuel_line_edit.text() == "0"
+    assert window.jump_line_edit.text() == "0"
+    assert window.jump_label.text() == "-"
+    assert window.thrust_line_edit.text() == "0"
+    assert window.thrust_label.text() == "-"
+
+    assert window.hull_hp_line_edit.text() == "2"
+    assert window.structure_hp_line_edit.text() == "2"
+    assert window.armour_line_edit.text() == "0"
+
+    assert window.discount.text() == "0"
+    assert window.cost_line_edit.text() == "2.500"
+
+
+def test_config_init(window):
+    """ Tests for base init of Config/Armor group """
+    assert window is not None
+
+    assert window.bridge_check.isChecked() is True
+    assert window.reflec_check.isChecked() is False
+    assert window.seal_check.isChecked() is False
+    assert window.stealth_check.isChecked() is False
+
+    assert window.meson_screen.isChecked() is False
+    assert window.nuclear_damper.isChecked() is False
+    assert window.fuel_scoop.isChecked() is False
+
+    assert window.hull_config_box.currentText() == "Standard"
+    assert window.sensors.currentText() == "Standard"
+    assert window.armor_combo_box.currentText() == "---"
+
+
+def test_edit_tonnage(window):
+    """ Tests for editing the tonnage of the ship """
+    assert window is not None
+
+    # Check setting the same tonnage
+    window.edit_tonnage()
+    assert window.spacecraft.tonnage == 100
+
+    # Set index to 200
+    window.tonnage_box.setCurrentIndex(1)
+    window.edit_tonnage()
+
+    # Check for values
+    assert window.spacecraft.tonnage == 200
+    assert window.spacecraft.get_remaining_cargo() == 190
+    assert window.spacecraft.get_total_cost() == 9.0
+    assert window.cost_line_edit.text() == "9.000"
+
+
+def test_tonnage_drive(window):
+    """
+    Test for automatically upgrading to lowest available drive on new tonnage
+    Tests for basic functionality of adding drives, as well
+    """
+    assert window is not None
+
+    # Add drives to ship
+    window.jump_line_edit.setText("A")
+    window.edit_jdrive()
+
+    window.thrust_line_edit.setText("A")
+    window.edit_mdrive()
+
+    window.pplant_line_edit.setText("A")
+    window.edit_pplant()
+
+    assert window.jump_label.text() == "A"
+    assert window.thrust_label.text() == "A"
+    assert window.pplant_label.text() == "A"
```

Comment:
This seems like a bad test

Reply:
The jump_labels are not the same as the line edits, but changed from the function call as it's a sublabel next to the line edit. It's a test seeing if the labels (and thus the underlying spacecraft object from which it pulls) goes through successfully.

Reply:
That's insane